enclave/attest_plugin.c

```diff
@@ -39,7 +39,32 @@ oe_result_t oe_unregister_attester_plugin(oe_attester_t* plugin)
         &attesters, (oe_attestation_role_t*)plugin);
 }
 
-static oe_result_t _wrap_with_header(
+oe_result_t oe_fill_attestation_header(
+    const oe_uuid_t* format_id,
+    const uint8_t* data,
+    size_t data_size,
+    oe_attestation_header_t* header)
+{
+    oe_result_t result = OE_UNEXPECTED;
+
+    if (!format_id || !data || !data_size || !header)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    // The header must prefix the data
+    if ((uint8_t*)header + sizeof(*header) != data)
```

Comment:
It seems like this method only exists to enforce this check, which requires that you've already laid out the buffer and calculated the offset to the data pointer, while reserving `sizeof(oe_attestation_header_t)` bytes at the start of it. Another way to look at it is that the check is a tautology: by definition of `oe_attestation_header_t` anything that comes after it is the `data`, why make someone pass you the pointer for it? 

Where it's currently being used outside of `oe_wrap_with_attestation_header` as in sgx/attester.c, it doesn't seem to do anything other than create a dev time failure path. It's probably easier to cast the buffer to `oe_attestation_header_t*` and just writing the `version`, `format_id` and `data_size` fields directly.

If the goal is to initialize constant fields to known good values (e.g. `magic` or `version`) consider defining a simple inline instead:
```c
inline void oe_set_attestation_header(
    uint8_t* buffer,
    size_t buffer_size,
    const oe_uuid_t* format_id,
    const size_t data_size);
```
That way you can check the minimum required size of `buffer_size` exceeds `data_size + sizeof(oe_attestation_header_t)` at least.

Reply:
Thanks! With the latest design, existing code will be used to handle the attestation headers.